src/articles/ArticlePreview.js

```diff
@@ -10,17 +18,74 @@ export default function ArticleOverview({
   dontShowImage,
   hasExtension,
 }) {
+  const [isOpen, setIsOpen] = useState(false);
+
   let topics = article.topics.split(" ").filter((each) => each !== "");
   let difficulty = Math.round(article.metrics.difficulty * 100) / 10;
 
+  function handleClose() {
+    setIsOpen(false);
+  }
+
+  function handleOpen() {
+    setIsOpen(true);
+  }
+
   function titleLink(article) {
     let open_in_zeeguu = (
       <Link to={`/read/article?id=${article.id}`}>{article.title}</Link>
     );
     let open_externally = (
-      <a target="_blank" rel="noreferrer" href={article.url}>
-        {article.title}
-      </a>
+      //TODO: Refactor into separate components and add styling to the modal and article titles
+      //Code related to the new redirection notification modal starts here
+      <>
+        <Modal open={isOpen} onClose={handleClose}>
+          <>
+            <MyBox>
+              <div>
+                <h1>
+                  You’re now leaving to&nbsp;a&nbsp;third&nbsp;party&nbsp;site.
```

Comment:
Maybe something more specific? E.g. ` You are being redirected to the original article page`